stonesoup/sensor/passive.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+
+from .base import Sensor3DCartesian
+from ..base import Property
+from ..models.measurement.nonlinear import CartesianToElevationBearing
+from ..types.array import CovarianceMatrix
+from ..types.detection import Detection
+
+
+class PassiveElevationBearing(Sensor3DCartesian):
+    """A simple passive sensor that generates measurements of targets, using a
+    :class:`~.CartesianToElevationBearing` model, relative to its position.
+
+    Note
+    ----
+    The current implementation of this class assumes a 3D Cartesian plane.
+
+    """
+
+    measurement_model = None
+
+    ndim_state = Property(
+        int,
+        doc="Number of state dimensions. This is utilised by (and follows in\
+            format) the underlying :class:`~.CartesianToElevationBearing`\
+            model")
+    mapping = Property(
+        [np.array], doc="Mapping between the targets state space and the\
+                        sensors measurement capability")
+    noise_covar = Property(CovarianceMatrix,
+                           doc="The sensor noise covariance matrix. This is \
+                                utilised by (and follow in format) the \
+                                underlying \
+                                :class:`~.CartesianToElevationBearing`\
+                                model")
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        self.measurement_model = CartesianToElevationBearing(
+            ndim_state=self.ndim_state,
+            mapping=self.mapping,
+            noise_covar=self.noise_covar,
+            translation_offset=self.position,
+            rotation_offset=self.orientation)
+
+    def measure(self, ground_truth, noise=None, **kwargs):
+        """Generate a measurement for a given state
+
+        Parameters
+        ----------
+        ground_truth : :class:`~.State`
+            A ground-truth state
+
+        Returns
+        -------
+        :class:`~.Detection`
+            A measurement generated from the given state. The timestamp of the\
+            measurement is set equal to that of the provided state.
+        """
+
+        measurement_vector = self.measurement_model.function(
+            ground_truth.state_vector, noise=noise, **kwargs)
+
+        return Detection(measurement_vector,
+                         measurement_model=self.measurement_model,
```

Comment:
I'd suggest creating a new instance of the measurement model here, rather than copying or using `self.measurement_model`. This way you remove the need for the init method, as well as the _set_position_ and _set_orientation_ methods.

I also think we should remove the _measurement_model_ property from the sensor, as so far none of the sensors use it.

Reply:
Changes made to remove _measurement_model_ from sensor.